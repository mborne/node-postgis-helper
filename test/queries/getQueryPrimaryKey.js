const expect = require("chai").expect;

const queries = require('../../src/queries');

describe("Test queries.getQueryPrimaryKey...", function () {

    it("should return an expected query for queries.getQueryPrimaryKey('sample','user')", function () {
        let query = queries.getQueryPrimaryKey('sample','user');
        //console.log(JSON.stringify(query,null,2));
        let expected = require('./expected/getQueryPrimaryKey.json');
        expect(query).to.deep.equals(expected);
    });


});

