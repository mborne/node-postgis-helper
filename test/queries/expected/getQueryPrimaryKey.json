{
    "text": "select\n\ttc.constraint_schema as schema,\n\ttc.constraint_name as table,\n\tkcu.column_name as column\nfrom information_schema.table_constraints tc\n    JOIN information_schema.key_column_usage AS kcu ON\n            tc.constraint_name = kcu.constraint_name\n        AND tc.constraint_schema = kcu.constraint_schema\nwhere tc.constraint_type = 'PRIMARY KEY'\n  and tc.table_schema = $1\n  and tc.table_name = $2\norder by kcu.ordinal_position",
    "values": [
        "sample",
        "user"
    ]
}